src/ui/Analytics/AnalyticsInformation.ts

```diff
@@ -3,12 +3,23 @@ import { findLastIndex } from 'underscore';
 import { Cookie } from '../../utils/CookieUtils';
 
 export class AnalyticsInformation {
+  private readonly visitorIdKey = 'visitorId';
+  private readonly clientIdKey = 'clientId';
+
   public get visitorId() {
-    return Cookie.get('visitorId') || null;
+    return localStorage.getItem(this.visitorIdKey) || Cookie.get(this.visitorIdKey) || null;
```

Comment:
In the project, we use a wrapper for local storage (`LocalStorageUtils`), since it's an operation that can throw in some browsers.

For example, in Safari in some special circumstances (iOS particularly), localStorage is completely de-activated/undefined.

You will also see that `LocalStorageUtils` always prefixes the keys with `coveo-`.

Reply:
Good call, changing to `LocalStorageUtils` 💯